### ISSUE TITLE
P2 a11y bug - episode list - 5 down key presses to move through the link element

### ISSUE THREAD

**Issue by jroebu14** [BBC-archive/psammead#4041]
**Describe the bug**
When testing with NVDA, when using the arrow down key, it is 5 presses of this key to move through the link element, this should be just one press of the down arrow key.

Adding a wrapping span with role text directly inside the link element may fix this.

This may also fix a similar issue on iPhone - P2 When testing this with VoiceOver on iPhone, when navigating via links, the links are several swipes to move through, they should just be one swipe. Adding a wrapping span with role text directly inside the link element may fix this.

**To Reproduce**
Steps to reproduce the behaviour:

**Expected behaviour**

**Screenshots**

**Desktop (please complete the following information):**

**Smartphone (please complete the following information):**

**Testing notes**

- [ ] This bug fix is expected to need manual testing.

**Additional context**
CC @greenc05

**Comment by LilyL0u** [comment]
This will hopefully also the fix the bug on Talkback:
P3 There is an announced comma in the navigation order that is isolated from and between the brand title and the next element (description or release date). There is another one between the description and the duration or the release date and the duration.
As these are part of the 5 tabs it takes to get through

**Comment by LilyL0u** [comment]
Tested on NVDA. The episode without a description reads the whole link in one down press. The episodes with a description reads it out in two down presses, but I think this is because with the description there are enough words that it is too long and it splits how it is read out. It does this with paragraph text too in general. I think this is fine and an expected part of the tool, and your fix has solved the problem.

On talkback, this has solved the commas problem too. 

However, perhaps due to a talkback quirk, each link is 2 key presses as it is read out twice.
On an episode with no desciption, the whole link is read out (twice with 2 key presses).  On an episode with a description, everything is read apart from the release date. It will read everything twice with 2 key presses, and THEN on the third key press it reads the release date.

The behaviour on http://2bitstudio.co.uk/code-tests/world-service/recent-episodes-audio-svg.html is:
The episode information is all read out ONCE (one press) on the episode with no description. But on the episode with a description, the episode information is read out twice (2 presses) and the release date is read out in the third press. This is the same behaviour as on Jonathan's fix.
Thoughts on this @greenc05 ?

**Comment by greenc05** [comment]
Thanks @LilyL0u sounds like the Talkback quirk could be due to your device? @DenisHdz any chance you could check this for us on Craig pretty please?

**Comment by jroebu14** [comment]
@LilyL0u episodes with a description should not read out the date because the date is not included in the anchor tag. Is this maybe what you are experiencing?

**Comment by DenisHdz** [comment]
This is how is announced on the Craig Android device using the default navigation:
- Episode with no description: `Audio, magazine de la culture, four Avril twenty twenty, twelve thirty, duration fifty nine o o, link` just once.
- Episode with description: `Audio, magazine de la culture, this is an episode with a title, duration fifty nine o o, link` just once.

**Comment by LilyL0u** [comment]
@jroebu14 looks like my device reading stuff twice is not a universal problem. Possibly just my, or a few types, of device

**Comment by LilyL0u** [comment]
> This is how is announced on the Craig Android device using the default navigation:
> 
>     * Episode with no description: `Audio, magazine de la culture, four Avril twenty twenty, twelve thirty, duration fifty nine o o, link` just once.
> 
>     * Episode with description: `Audio, magazine de la culture, this is an episode with a title, duration fifty nine o o, link` just once.

Thanks Denis and Craig

**Comment by gavinspence** [comment]
Can we close this issue then? :)

**Comment by LilyL0u** [comment]
I'll close it
